ev3dev/helper.py

```diff
@@ -146,6 +147,171 @@ class MediumMotor(ev3dev.auto.MediumMotor, MotorMixin):
     pass
 
 
+class MotorSet(object):
+    """
+    motor_specs is a dictionary such as
+    {
+        OUTPUT_A : LargeMotor,
+        OUTPUT_B : MediumMotor,
+    }
+    """
+
+    def __init__(self, motor_specs, desc=None):
+
+        for motor_port in motor_specs.keys():
+            if motor_port not in OUTPUTS:
+                log.error("%s in an invalid motor, choices are %s" % (motor_port, ', '.join(OUTPUTS)))
+                sys.exit(1)
+
+        self.motors = OrderedDict()
+        for motor_port in sorted(motor_specs.keys()):
+            motor_class = motor_specs[motor_port]
+            self.motors[motor_port] = motor_class(motor_port)
+
+        self.desc = desc
+        self.verify_connected()
+
+    def __str__(self):
+
+        if self.desc:
+            return self.desc
+        else:
+            return self.__class__.__name__
+
+    def verify_connected(self):
+        for motor in self.motors.values():
+            if not motor.connected:
+                log.error("%s: %s is not connected" % (self, motor))
+                sys.exit(1)
+
+    def set_args(self, **kwargs):
+        motors = kwargs.get('motors', self.motors.values())
+
+        for motor in motors:
+            for key in kwargs:
+                if key != 'motors':
+                    setattr(motor, key, kwargs[key])
+
+    def _run_command(self, **kwargs):
+        motors = kwargs.get('motors', self.motors.values())
+
+        for motor in motors:
+            for key in kwargs:
+                if key not in ('motors', 'commands'):
+                    log.info("%s: %s set %s to %s" % (self, motor, key, kwargs[key]))
+                    setattr(motor, key, kwargs[key])
+
+        for motor in motors:
+            motor.command = kwargs['command']
+            log.info("%s: %s command %s" % (self, motor, kwargs['command']))
+
+    def run_forever(self, **kwargs):
+        kwargs['command'] = ev3dev.auto.LargeMotor.COMMAND_RUN_FOREVER
+        self._run_command(**kwargs)
```

Comment:
One thing I am not sure about is the logic of these `run_*` functions. Here you are pushing command into kwargs, and then filter it out in [`_run_command` function](https://github.com/dwalton76/ev3dev-lang-python/blob/f928075be9f7079d093cc44c04fb0caeb3fff2f3/ev3dev/helper.py#L198-L202). This cost a string comparison per key in kwargs, which could be negligible with respect to file acesses, but still is not free.

Why not create `_set_args(self, **kwargs)` function, and replace the two lines above with
```python
self._set_args(**kwargs)
self._run_command(ev3dev.auto.LargeMotor.COMMAND_RUN_FOREVER)
```
?

Reply:
There is a set_args() (see line 187) and the user can do it this way if they want but they don't have to.

The price of the string compare feels worth it to me, it let me clean up this code a good bit by moving almost all of the logic to _run_command().

Reply:
Well, I don't have strong feelings about this, so feel free to merge.